### PR TITLE
FSET-1374: Candidate needs to submit their analysis exercise while they are at the assessment centre

### PR DESCRIPTION
**Pull request by henricook**

### CODE REVIEW

File: app/services/allocation/CandidateAllocationService.scala [hunk 1]
```diff
@@ -74,6 +76,23 @@ trait CandidateAllocationService extends EventSink {
     candidateAllocationRepo.allocationsForSession(eventId, sessionId).map { a => exchange.CandidateAllocations.apply(a) }
   }
 
+  def getSessionsForApplication(applicationId: String, sessionEventType: EventType): Future[List[Event]] = {
+    for {
+      allocations <- candidateAllocationRepo.allocationsForApplication(applicationId)
+      events <- eventsService.getEvents(allocations.map(_.eventId).toList, sessionEventType)
+    } yield {
+      Logger.warn("ALL = " + allocations)
+      Logger.warn("Events = " + events)
```
Comment (kjharkin) [inline app/services/allocation/CandidateAllocationService.scala]: These logging entries should be removed.

Comment (henricook) [inline app/services/allocation/CandidateAllocationService.scala]: :blush: - apologies

File: app/services/assessmentcentre/AssessmentCentreService.scala [hunk 2]
```diff
@@ -19,15 +19,21 @@ package services.assessmentcentre
...
 
 import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 
 object AssessmentCentreService extends AssessmentCentreService {
   val applicationRepo = repositories.applicationRepository
   val assessmentCentreRepo = repositories.assessmentCentreRepository
+
+  case class CandidateAlreadyHasAnAnalysisExerciseException(message: String) extends Exception(message)
+  case class CandidateAlreadyHasNoAnalysisExerciseException(message: String) extends Exception(message)
```
Comment (kjharkin) [inline app/services/assessmentcentre/AssessmentCentreService.scala]: Is this named correctly (e.g. should it be "CandidateHasNoAnalysisExerciseException") ?

File: it/repositories/fileupload/FileUploadRepositorySpec.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      storeAndRetrieveFile
+    }
+  }
+
+  "retrieve" must {
+    "retrieve a file by id" in {
+      storeAndRetrieveFile
+    }
+  }
+}
```
Comment (kjharkin) [inline it/repositories/fileupload/FileUploadRepositorySpec.scala]: I don't like this style - IMHO I would like tests to be readable - DRY doesn't apply

**Pull request merged**